src/engine/renderer/glsl_source/generic2D_fp.glsl

```diff
@@ -0,0 +1,26 @@
+/*
+===========================================================================
+Copyright (C) 2006-2011 Robert Beckebans <trebor_7@users.sourceforge.net>
+
+This file is part of XreaL source code.
+
+XreaL source code is free software; you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation; either version 2 of the License,
+or (at your option) any later version.
+
+XreaL source code is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with XreaL source code; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+===========================================================================
+*/
+
+/* generic2D_fp.glsl */
+
+#define GENERIC_2D 1
+
```

Comment:
There's nothing here?

Reply:
Apparently it gets merged with generic_fp.glsl

Reply:
You're not actually compiling this, only the "generic" one again.
```
GLShader_generic2D::GLShader_generic2D( GLShaderManager *manager ) :
	GLShader( "generic", ATTR_POSITION | ATTR_TEXCOORD | ATTR_QTANGENT, manager ),
```

Reply:
An internal `generic2D` compiles `generic_fp.glsl` using `generic2D_fp.glsl` as concatenated header:

```c++
GLShader_generic2D::GLShader_generic2D( GLShaderManager *manager ) :
	GLShader( "generic", ATTR_POSITION | ATTR_TEXCOORD | ATTR_QTANGENT, manager ),
	[…]

void GLShader_generic2D::BuildShaderFragmentLibNames( std::string& fragmentInlines )
{
	fragmentInlines += "generic2D";
}
```

Later I want to implement a mechanism to set a define without needing a dedicated file but this should work.